File: YouCafeteria/src/main/java/org/rabie/youcafeteria/service/StockService.java
package org.rabie.youcafeteria.service;

import org.rabie.youcafeteria.domain.Stock;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Service;

@Service
public interface StockService {

    Stock saveStock(Stock stock);
    Stock updateStock(Stock stock);
    void deleteStock(String name);
    Stock getStock(String name);
    Page<Stock> getAllStocks(int page, int size);
    Stock save(Stock stock, boolean isUpdate);

}

File: youcafeteria-angular/src/app/service/stock.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class StockService {
  private url = environment.apiUrl + 'stock';

  constructor(private http: HttpClient) {}

  getAllStocks(page: number = 0, size: number = 100): Observable<any> {
    const params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString());
    return this.http.get(`${this.url}/all`, { params });
  }

  getStock(name: string): Observable<any> {
    return this.http.get(`${this.url}/details/${name}`);
  }

  createStock(data: any): Observable<any> {
    return this.http.post(`${this.url}/create`, data);
  }

  updateStock(data: any): Observable<any> {
    return this.http.put(`${this.url}/update`, data);
  }

  deleteStock(name: string): Observable<any> {
    return this.http.delete(`${this.url}/delete/${name}`);
  }
}
